lib/visitor.js

```diff
@@ -280,19 +282,44 @@ class Visitor {
             }, '};') // end of generated class
         }, '}') // end of aspect
 
+        // DRAFT CLASSES
+        this.#printDraftClasses(fq, buffer, identSingular(clean), identAspect(clean))
+
         // CLASS WITH ADDED ASPECTS
         file.addImport(baseDefinitions.path)
         docify(entity.doc).forEach(d => { buffer.add(d) })
-        buffer.add(`export class ${identSingular(clean)} extends ${identAspect(toLocalIdent({clean, fq}))}(${baseDefinitions.path.asIdentifier()}.Entity) {${this.#staticClassContents(clean, entity).join('\n')}}`)
+        buffer.add(`export class ${identSingular(clean)} extends ${identAspect(toLocalIdent({clean, fq}))}(${baseDefinitions.path.asIdentifier()}.Entity) {${this.#staticClassContents(fq, clean, entity).join('\n')}}`)
         this.contexts.pop()
     }
 
     /**
+     * Prints draft classes for draft enabled
+     * @param {string} fq - FQN of entity
+     * @param {Buffer} buffer - buffer to write exported drafts classes to
+     * @param {string} singular - singular name of parent entity
+     * @param {string} aspectFunction - class creator function of parent entity
+     */
+    #printDraftClasses(fq, buffer, singular, aspectFunction) {
+        if (!isDraftEnabled(fq)) return
+        const singularDraft = `${singular}Draft`
+        // add singular draft
+        buffer.addIndentedBlock(`export class ${singularDraft} extends ${aspectFunction}(__.DraftEntity) {`, () => {
```

Comment:
I am always very, very sceptical when we try to introduce identifiers in the types that are not actually in the CSN. It's already sometimes an issue with automatic inflection, and has recently [become an issue with services](https://github.com/cap-js/cds-typer/issues/288) that are now also generated into the _index.ts_, which formerly represented the service itself. (I promise you there will be someone who has an entity `Foo` _and_ another entity `FooDraft` 😬)

Reply:
I am totally with you. The suffix was merely a suggestion and I am not married to it 😊. We could make the name configurable via annotations (`@draftSingular`, `@draftPlural`), or globally via CLI option (`draft_class_suffix`) or another approach that would allow the user to fix the types if their model falls into an edge case.

If you do not like the dedicated class approach for the drafts at all, we can skip them entirely and just enhance the type behind the `.drafts` property to include the properties of a draft entity.

```ts
// /@cds-models/_/index.ts
type DraftEntity<T> = T & {
    IsActiveEntity?: boolean | null
    HasActiveEntity?: boolean | null
    HasDraftEntity?: boolean | null
    DraftAdministrativeData_DraftUUID?: string | null
}
export type DraftOf<T> = { new(...args: any[]): DraftEntity<T> }
export type DraftsOf<T> = typeof Array<DraftEntity<T>>

// /@cds-models/CatalogService/index.ts
import * as __ from './../_';

export class Book extends _BookAspect(__.Entity) {static drafts: __.DraftOf<Book>}
export class Books extends Array<Book> {static drafts: __.DraftsOf<Book>}
```

Reply:
> just enhance the type behind the .drafts property to include the properties of a draft entity.

I think this would be the most preferable approach. Introducing CSN annotations always has to be approved internally first. And I'd prefer to introduce CLI options sparingly. So enhancing the type sounds very appealing to me.

Reply:
I made the changes. One downside of omitting the dedicated draft classes was the advantage of easy type casts.

e.g.

```ts
INSERT.into(Books.drafts).entries({IsActiveEntity: true} as BookDraft);

let bookDraft = {} as  BookDraft;
bookDraft.IsActiveEntity = true
```

So the user still can do this in a rather easy way I exported the new `DraftEntity` type from the generated `/_/index.ts` file.

```ts
INSERT.into(Books.drafts).entries({IsActiveEntity: true} as DraftEntity<Book>);

let bookDraft = {} as  DraftEntity<Book>;
bookDraft.IsActiveEntity = true
```

Reply:
That sounds reasonable and totally doable for users when they want to reuse this type. 👍